### PR TITLE
Server::handleChat: Add missing & to prevent unnecessary string copy

### PR DESCRIPTION
**Pull request by ClobberXD**

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -2859,7 +2859,7 @@ void Server::handleChatInterfaceEvent(ChatEvent *evt)
 }
 
 std::wstring Server::handleChat(const std::string &name, const std::wstring &wname,
-	std::wstring wmessage, bool check_shout_priv, RemotePlayer *player)
+	std::wstring &wmessage, bool check_shout_priv, RemotePlayer *player)
```
Comment (nerzhul) [inline src/server.cpp]: no, you will modify original string as it's not constant

Comment (ClobberXD) [reply src/server.cpp]: Yes, which is why I haven't made it a `const`. That being said, does our code-style even allow non-const references? :thinking:

Comment (nerzhul) [reply src/server.cpp]: it's not a problem of code style, you will modify the calling passed argument, which is not what we want here.

Comment (ClobberXD) [reply src/server.cpp]: I understand, thanks

**Pull request closed without merging**